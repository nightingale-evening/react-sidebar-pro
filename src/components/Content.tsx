import React from 'react';
import styled from 'styled-components';
import classnames from 'classnames';

export type ContentProps = React.HTMLAttributes<HTMLHtmlElement>;

const StyledContent = styled.main`
  flex: 1;
`;

export const Content: React.FC<ContentProps> = React.memo(({ children, className, ...rest }) => {
  return (
    <StyledContent
      data-testid="content-test-id"
      className={classnames('content', className)}
      {...rest}
    >
      {children}
    </StyledContent>
  );
});
