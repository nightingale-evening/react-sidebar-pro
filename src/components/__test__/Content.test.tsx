import React from 'react';
import { customRender, screen } from '../../utils/testUtils';
import { Content } from '../Content';

describe('Content', () => {
  it('basic snapshot ', () => {
    const { container } = customRender(<Content>Content</Content>);

    expect(container.firstChild).toMatchSnapshot();
  });

  it('should initialize Content correctly', async () => {
    customRender(<Content>Content</Content>);
    const contentElem = screen.getByText('Content');

    expect(contentElem).toBeInTheDocument();
    expect(contentElem).toHaveClass('content');
    expect(contentElem).toHaveStyle({
      flex: '1',
    });
  });
});
