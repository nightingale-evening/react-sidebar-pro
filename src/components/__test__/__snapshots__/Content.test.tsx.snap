// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Content basic snapshot  1`] = `
<main
  class="sc-bdvvtL gZywae content"
  data-testid="content-test-id"
>
  Content
</main>
`;
